instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteAlternativesTest.kt

```diff
@@ -113,11 +116,149 @@ class RouteAlternativesTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::cla
         }
         countDownLatch.await()
 
+        // Verify alternative routes events were triggered.
+        assertEquals(2, routes.size)
+        assertTrue(mapboxNavigation.getNavigationRoutes().isNotEmpty())
+        assertNotNull(firstAlternative.routeProgress)
+        assertNotNull(firstAlternative.alternatives)
+        firstAlternative.alternatives!!.forEach {
+            assertFalse(routes.contains(it))
+        }
+    }
+
+    @Test
+    fun alternative_routes_observer_is_not_triggered_for_set_routes() {
+        // Prepare with alternative routes.
+        setupMockRequestHandlers(coordinates)
+        val routes = requestDirectionsRouteSync(coordinates)
+
+        // Play the slower alternative route.
+        runOnMainSync {
+            mapboxNavigation.historyRecorder.startRecording()
+            mockLocationReplayerRule.playRoute(routes.first())
+            mapboxNavigation.startTripSession()
+        }
+
+        // Wait for enhanced locations to start and then set the routes.
+        val firstLocationIdlingResource = FirstLocationIdlingResource(mapboxNavigation)
+        firstLocationIdlingResource.firstLocationSync()
+        runOnMainSync {
+            mapboxNavigation.setRoutes(routes)
+        }
+
+        // The alternative route is missed, so we expect the route to be removed.
+        val firstAlternative = RouteAlternativesIdlingResource(mapboxNavigation)
+        firstAlternative.register()
+        mapboxHistoryTestRule.stopRecordingOnCrash("alternatives failed") {
+            Espresso.onIdle()
+        }
+        firstAlternative.unregister()
+        assertTrue(firstAlternative.calledOnMainThread)
```

Comment:
The test says you want to check that the observer is not triggered. But here you check that the flag is true and it is set to true only when it's triggered. I'm confused.
Why do you need this check at all?

Reply:
the test checks that the observer is not triggered for `MapboxNavigation#setRoutes`, but waits for the very first trigger (that happens later). 

>Why do you need this check at all?

this check was added in the previous test (see `expect_initial_alternative_route_removed_after_navigating_past`), here we can re-use it for free without additional effort.

Reply:
> check that the observer is triggered

could you point out what you mean?
```
        assertTrue(firstAlternative.calledOnMainThread)
```
and 
```
        assertNotNull(firstAlternative.routeProgress)
        assertNotNull(firstAlternative.alternatives)
```
checks are different

>wait until we stop recording (btw, what's recording have to do with the alternatives?)

the recoding is storing history recording files to profile tests if something goes wrong

Reply:
> check that the observer is triggered

I meant `assertTrue(firstAlternative.calledOnMainThread)`. It's only true when triggered. But OK, I see what you mean. You want to check that it's triggered but not for setRoutes. Then I just find the `calledOnMainThread` check redundant here.

> the recoding is storing history recording files to profile tests if something goes wrong

Oh, cool, I didn't use this in my tests because I wanted to keep them as simple as possible. BTW, can this be automised? I mean it would be cool if history was recorded for each test under-the-hood so that we don't overcomplicate the test body. WDYT? Should I cut a ticket for discussion or is it something that you've already discussed and decided it's not possible/we don't want that?

Reply:
>Then I just find the calledOnMainThread check redundant here.

I see what you mean 👍 
- don't see anything bad to verify threads in every place where it might be needed (we work with interfaces and we're agnostic to what is happening under an interface if switching to a new thread depends on any logic);
- I'd say we need a common approach to verify threads in tests. I think kind of rule of util which we might inject into a code block of instrumentation tests and verify threads. As we haven't had anything like this let's keep the same here and try to create something more generic. I cut the issue https://github.com/mapbox/navigation-sdks/issues/1970

> I mean it would be cool if history was recorded for each test under-the-hood so that we don't overcomplicate the test body. WDYT? Should I cut a ticket for discussion or is it something that you've already discussed and decided it's not possible/we don't want that?

yeah, agree. Great idea. Could you cut a ticket?

Reply:
`navigation-sdks#1971`